### ISSUE TITLE
[REQUEST] please support ipv6 for masternode 

### ISSUE THREAD

**Issue by encryptblockr** [dashpay/dash#1931]
Current masternode setup only supports ipv4
As one of the leading crypto currencies on the market, will be great to support ipv6 

Thanks

**Comment by UdjinM6** [comment]
This was discussed many times on dash.org/forum.

TL;DR: not sooner the IPv6 adoption rises to some meaningful levels, see https://www.google.com/intl/en/ipv6/statistics.html etc. OR when we drop everything that requires direct IP access to masternodes (like mixing) OR when we switch to some kind of custom routing for such cases (which is going to be much slower and require much more traffic IMO).

**Comment by encryptblockr** [comment]
@UdjinM6 

other MN coins using ipv6
so i don't get where the complexity is coming from?
should be able to support ipv4, ipv6, and tor

**Comment by schinzelh** [comment]
Did you even take the time to read into the discussions WHY masternode entries other than IPv4 are not supported? 

The complexity comes from the fact the you are not able to properly route between the different networks without the help of another overlay network - which none of the coins has afaik.

E.g. tell me: How is a IPv4-only connected wallet supposed to connect to a Tor listed masternode during mixing?

We are well aware that 150 MN coins copied our masternode code and re-enabled IPv6/Tor list entries, but they do not actually use these entries for something useful. 

But maybe you know something we don't know, so: What "other MN coin" is actually **using** the nodes for something different than locking up coins and have them idling around?

**Comment by nmarley** [comment]
Technical discussion on the issue here: https://github.com/dashpay/dash/pull/1065

TL;DR: the Dash team didn't _want_ to have to disable ipv6 for MNs, but there are technical reasons this change was made, and it was for the health of the network (as @schinzelh said, ipv4 only nodes couldn't connect to ipv6 nodes). With ipv4-only as it is now, all MNs are able to provide a benefit to the _entire_ network (not just ipv6 nodes).

**Comment by nmarley** [comment]
Closing issue as it's a duplicate and this is a known, intentional Dash network architecture decision.
